inausoft.netCLI.all/inausoft.netCLI.Tests/RootCommandHandlerTests.cs

```diff
@@ -93,6 +97,89 @@ public void TestMethod2()
             Assert.AreEqual(0, mockCommand1Handler.LastRunParameters.IntOption);
         }
 
+        [TestMethod]
+        public void Command2WithoutParameters_ReturnsCorrectCommand()
+        {
+            //Arrange
+            var args = new string[]
+                            {
+                                "command2",
+                            };
+
+            //Act
+            var mockCommand1Handler = new MockCommand1Handler();
+            var mockCommand2Handler = new MockCommand2Handler();
+            var rootCommandHandler = new RootCommandHandler(new ICommandHandler[] { mockCommand2Handler, mockCommand1Handler });
+
+            var result = rootCommandHandler.Run(args);
+
+            //Assert
+            Assert.AreEqual(0, result);
+
+            Assert.IsNotNull(mockCommand2Handler.LastRunParameters);
+        }
+
+        [ExpectedException(typeof(InvalidCommandException))]
+        [TestMethod]
+        public void Command3WithoutRequiredParameter_ThrowsExcception()
+        {
+            //Arrange
+            var args = new string[]
+                            {
+                                "command3",
+                                  "--OptionalArgument", "true",
+                            };
+
+            //Act
+            var mockCommand3Handler = new MockCommand3Handler();
+            var rootCommandHandler = new RootCommandHandler(new ICommandHandler[] { mockCommand3Handler });
+
+            var result = rootCommandHandler.Run(args);
+            //Assert with exception
+        }
+
+        [TestMethod]
+        public void Command3WithoutRequiredParameter_RequiredParameterShouldReturnNull()
+        {
+            //Arrange
+            var args = new string[]
+                            {
+                                "command3",
+                                  "--OptionalArgument", "true",
+                            };
+
+            //Act
+            var mockCommand3Handler = new MockCommand3Handler();
+            var rootCommandHandler = new RootCommandHandler(new ICommandHandler[] { mockCommand3Handler });
+
+            var result = rootCommandHandler.Run(args);
+            Assert.IsTrue(mockCommand3Handler.LastRunParameters.OptionalArgument);
+            //Because of boolean value type, we still think that argument was set. Maybe we should use only nullable types? 
```

Comment:
Not true. This is user choice - either explicit or not. While writing command - user is definig defaul values. See example:
```
[Command("command3")]
    class Command3
    {
        [Option("RequiredArgument")]
        public bool RequiredArgumentWithDefaultTrueValue { get; set; }

        public Command3()
        {
              RequiredArgumentWithDefaultTrueValue = true;
        }
    }
```
Not providing options will result in initializing command with default values - which I think is desirable.

Reply:
I disagree. You could specified that command should have mandatory arguments (like --targetversion of  type int), that when wasn't provided, it would return 0, which seems like invalid target version (it just an example) 
What I mean that null is always incorrect value, that should ends with throwing exception on argument usage, in opposite with default values of datatypes. Therefore, not provided parameters should be null.

Reply:
You can easily achive this with proposed implementation and example above. It's consumer choice to implement properties as nullable and throw `ArgumentNullException` if needed it is possible, same as having default values.

Reply:
Agreed